### ISSUE TITLE
default_connection doesn't seem to be using default_connection_options.

### ISSUE THREAD

**Issue by herberzt** [lostisland/faraday#698]
We are currently using the acme-client gem (https://github.com/unixcharles/acme-client) to manage some SSL certificates. In doing so we have come to find that the `default_connection` doesn't honor `default_connection_options` and they must be explicitly set through `Faraday::Connection.new()`. In order to work around this we had to do the following:

```
  Faraday.default_connection_options = {request: { open_timeout: 5, timeout: 5 }}
  Faraday.default_adapter = :httpclient
  Faraday.default_connection = Faraday::Connection.new(nil, Faraday.default_connection_options)
```
The recommended way to do this through the acme-client gem was the following:
```
client = Acme::Client.new(private_key: private_key, endpoint: endpoint, connection_options: { request: { open_timeout: 5, timeout: 5 } })
```

ACME uses calls `Faraday.head` which uses `default_connection` which currently doesn't seem to be using `default_connection_options`.

**Comment by iMacTia** [comment]
Hi @herberzt and thanks for reporting this.
I think you spot a regression bug introduced with one of the recent commits.
We moved the `default_connection_options` merging to `Faraday.new` ignoring the `default_connection` method.
I'll get this fixed in the next release.

**Comment by iMacTia** [comment]
@herberzt Sorry it took me some time, but a PR is now available.
I would appreciate your input on it 😃 
Moreover, would you be able to test your app with the `fix/#968` branch and let me know if your issue is now solved? Thanks
